src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -41,9 +43,61 @@
 #define KDATEONLY_FORMAT    "%Y-%m-%d"
 #define QDATEONLY_FORMAT    "yyyy-MM-dd"
 #define KLONGTZ_FORMAT      "%:Z"
+#define RFC5545_KDATETIME_FORMAT "%Y%m%dT%H%M%SZ"
+#define RFC5545_KDATETIME_FORMAT_NTZC "%Y%m%dT%H%M%S"
+#define RFC5545_QDATE_FORMAT "yyyyMMdd"
 
 namespace {
 
+
+void errorDumpStr(const QString &str)
+{
+    // Dump the entire string to the log.
+    // Note that the log cannot handle newlines,
+    // so we separate the string into chunks.
+    QString chunk;
+    for (int i = 0; i < str.size(); ++i) {
+        if (str.at(i) == '\n') {
+            if (chunk.size()) {
+                SOCIALD_LOG_ERROR(chunk);
+                chunk.clear();
+            }
+        } else {
+            chunk += str.at(i);
```

Comment:
Yeah, I'll use QString::split() and a foreach loop.
I don't want to try to handle it in the macro, for simplicity, as it's just a thin wrapper around qWarning()<<msg; via the Buteo macros.